### ISSUE TITLE
KML/KMZ not loadable outside of Google Earth Pro

### ISSUE THREAD

**Issue by Saijin-Naib** [OSGeo/gdal#1965]
## Expected behavior and actual behavior.
File(s) should be readable by QGIS/GDAL and other tools outside of just Google Earth Pro.
Files were genereated by the NYS DEC.

## Steps to reproduce the problem.
Attempt to load the data as spatial data in QGIS or other tools that rely upon GDAL for KML/KMZ handling, and you get a tabluar file with a broken layer name.

## Operating system
Win10 1909 x64

## GDAL version and provenance
v2.4.1 (QGIS 3.10.0 distributed)

The below attached KMZ files have been re-saved/exported via Google Earth Pro and still can not be read properly.
[KMLs.zip](https://github.com/OSGeo/gdal/files/3785461/KMLs.zip)

Below is an example of one of the doc.kml files, with a malformed Document Name object.
```
<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2" xmlns:gx="http://www.google.com/kml/ext/2.2" xmlns:kml="http://www.opengis.net/kml/2.2" xmlns:atom="http://www.w3.org/2005/Atom">
<Document>
	<name>&lt;font face=&quot;helvetica&quot; font color=&quot;#2E7B1B&quot;&gt;DEC Campgrounds&lt;/font&gt;</name>
	<NetworkLink>
		<name>DEC Campgrounds</name>
		<open>1</open>
		<description>Data is automatically updated whenever the file is opened. Right-click and select &quot;Refresh&quot; for additional updates.</description>
		<refreshVisibility>1</refreshVisibility>
		<flyToView>1</flyToView>
		<Link>
			<href>http://www.dec.ny.gov/maps/campgrounds.kmz</href>
		</Link>
	</NetworkLink>
</Document>
</kml>
```

Furthermore, it would appear that these are just linking to data hosted on the NYSDEC server and are live-updated when loaded, at least in Google Earth.

**Comment by rouault** [comment]
There is a strong impedance mismatch between the KML model and GDAL/OGR ones. KML can be seen both as a way to both encode a project and store data for visualization. GDAL has a hard time figuring out if the KML can be considered as a raster or a vector file. For the above file, one should open the <href>http://www.dec.ny.gov/maps/campgrounds.kmz</href> to figure out what it is.

**Comment by Saijin-Naib** [comment]
So, this file is not special in any way and this behavior is expected (from GDAL's side)?

**Comment by rouault** [comment]
yes, it malfunctions as expected... I guess that a dedicated QGIS plugin (if one doesn't yet exist) to convert a KML/KMZ to a QGIS project could limit the amount of loss.

**Comment by Saijin-Naib** [comment]
https://github.com/NationalSecurityAgency/qgis-kmltools-plugin
One exists, but appears to lean on GDAL, so there is no appreciable difference in function.
